examples/src/main/java/tachyon/examples/BasicNonByteBufferOperations.java

```diff
@@ -37,20 +44,30 @@
  * </p>
  * <p>
  * This example also let users play around with how to work with files a bit more. The
- * {@link tachyon.client.ReadType} is something that can be set, as well as ability to delete file
- * if exists.
+ * {@link tachyon.client.TachyonStorageType} is something that can be set, as well as ability to
+ * delete file if exists.
  * </p>
  */
 public final class BasicNonByteBufferOperations implements Callable<Boolean> {
+  // private access to the reinitializer of ClientContext
+  private static ClientContext.ReinitializerAccesser sReinitializerAccesser =
+      new ClientContext.ReinitializerAccesser() {
+        @Override
+        public void receiveAccess(ClientContext.PrivateReinitializer access) {
+          sReinitializer = access;
+        }
+      };
+  private static ClientContext.PrivateReinitializer sReinitializer;
   private final TachyonURI mMasterLocation;
   private final TachyonURI mFilePath;
-  private final WriteType mWriteType;
-  private final ReadType mReadType;
+  private final TachyonStorageType mWriteType;
+  private final TachyonStorageType mReadType;
   private final boolean mDeleteIfExists;
   private final int mLength;
 
   public BasicNonByteBufferOperations(TachyonURI masterLocation, TachyonURI filePath,
-      WriteType writeType, ReadType readType, boolean deleteIfExists, int length) {
+      TachyonStorageType writeType, TachyonStorageType readType, boolean deleteIfExists,
```

Comment:
because for original example, read & write can be set separately, so I just tried to follow the existing pattern, i.e. user can set STORE for read operation, NO_STORE for write operation, or NO_STORE for read operation, STORE for write operation. I guess from original example these are all feasible. Maybe we should change the name to  tachyonStorageTypeForWrite and tachyonStorageTypeForRead?